### DISCUSSION TITLE
Do we need to use DynamicModel?

### DISCUSSION THREAD

**Discussion by lukepighetti** [pocketbase/pocketbase#5329] [Q&A]
DynamicModel is a great tool, but what if I just want to always return the entire contents of a row as a map without specifying the fields in my JS function?

**Comment by ganigeorgiev** [comment]
The JSVM (goja) DB APIs are direct translation from Go and you need to specify the fields you are retrieving in a `DynamicModel` because otherwise we don't know into what to unmarshal the fetched row result.

We don't do any checks for the related table column types which often may not be even known before evaluation (e.g. conditional column expression) and for SQLite you can't rely on it since a single column technically could store values from different types (even for "strict" tables).

If you are operating with records then you don't have to explicitly type the columns and can make use of the record helpers - https://pocketbase.io/docs/js-records/#fetch-records (the columns value type is determined based on the collection fields).
